src/main/PDF/Services/CrudServices/ImageToPDFService.java

```diff
@@ -0,0 +1,107 @@
+import Config.Service;
+import PDF.PdfMessage;
+import org.apache.commons.io.output.ByteArrayOutputStream;
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.pdmodel.PDPage;
+import org.apache.pdfbox.pdmodel.PDPageContentStream;
+import org.apache.pdfbox.pdmodel.graphics.image.LosslessFactory;
+import org.apache.pdfbox.pdmodel.graphics.image.PDImageXObject;
+
+import javax.imageio.ImageIO;
+import java.awt.geom.AffineTransform;
+import java.awt.image.AffineTransformOp;
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+public class ImageToPDFService implements Service {
+    public static final int PDF_HEIGHT = 792;
+    public static final int PDF_WIDTH = 612;
+    private InputStream fileStream;
+
+    public ImageToPDFService(InputStream fileStream) {
+        this.fileStream = fileStream;
+    }
+
+    @Override
+    public Message executeAndGetResponse() {
+        if (fileStream == null) {
+            return PdfMessage.INVALID_IMAGE;
+        } else {
+            try {
+                BufferedImage bimg = ImageIO.read(fileStream);
+                if (bimg == null) return PdfMessage.INVALID_IMAGE;
+
+                fileStream = convertImageToPDF(bimg);
+            } catch (IOException exception) {
+                return PdfMessage.INVALID_IMAGE;
+            }
+        }
+        return PdfMessage.SUCCESS;
+    }
+
+    public InputStream getFileStream() {
+        return fileStream;
+    }
+
+    private InputStream convertImageToPDF(BufferedImage bimg) throws IOException {
+        PDDocument document = new PDDocument();
+        PDPage page = new PDPage();
+        document.addPage(page);
+
+        // Get new dimensions of rotated & scaled image
+        float imageWidth = bimg.getWidth();
+        float imageHeight = bimg.getHeight();
+
+        // Rotate Image
+        if (imageWidth > imageHeight) {
+            bimg = this.rotateImage(bimg);
+
+            float tmp = imageWidth;
+            imageWidth = imageHeight;
+            imageHeight = tmp;
+        }
+
+        // Scale Image
+        float finalWidth, finalHeight;
+        if (imageWidth / PDF_WIDTH > imageHeight / PDF_HEIGHT) {
+            finalWidth = PDF_WIDTH;
+            finalHeight = imageHeight * PDF_WIDTH / imageWidth;
+        } else {
+            finalHeight = PDF_HEIGHT;
+            finalWidth = imageWidth * PDF_HEIGHT / imageHeight;
+        }
+
+        PDImageXObject imageXObject = LosslessFactory.createFromImage(document, bimg);
+        PDPageContentStream contentStream = new PDPageContentStream(document, page);
+        contentStream.drawImage(imageXObject, 0, 0, finalWidth, finalHeight);
+        contentStream.close();
+
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        document.save(out);
+        document.close();
+        ByteArrayInputStream inputStream = new ByteArrayInputStream(out.toByteArray());
```

Comment:
I would rename this to convertedInputStream just to make it more clear that this is the output of your rotation

Reply:
Done